tests/test_base.cpp

```diff
@@ -562,7 +580,7 @@ std::string server_node::config_path() const
 	return m_path;
 }
 
-server_config server_node::config() const
+server_config &server_node::config()
```

Comment:
Also, you break the whole logic of the test, since tests are repeatable and independent.
If it is not possible (although quite desirable), you should at least tell others what and how your tests are supposed to work and what do the test, what they expect and how this can happen in real life.